### PR TITLE
Make sure old files are not picked up

### PR DESCRIPTION
**Pull request by mraspaud**
This PR saves the state of the directory to make sure only new files are picked.

Also adds the possibility to use a configured regex to parse new filenames from the script log.

### CODE REVIEW

File: pytroll_runner/__init__.py [hunk 1]
```diff
@@ -47,12 +49,22 @@ def parse_args(args=None):
...
             message = generate_message_from_expected_files(publisher_config, mda)
             pub.send(message)
 
 
+def check_existing_files(publisher_config):
+    """Check for previously generated files."""
+    global FILES_ALREADY_THERE
+    filepattern = publisher_config["expected_files"]
+    files = set(glob(filepattern))
+    FILES_ALREADY_THERE = files
```
Comment (pnuu) [inline pytroll_runner/__init__.py:65]:
```suggestion
    filepattern = publisher_config["expected_files"]
    return set(glob(filepattern))
```

File: pytroll_runner/__init__.py [hunk 2]
```diff
@@ -47,12 +49,22 @@ def parse_args(args=None):
 def run_and_publish(config_file):
     """Run the command and publish the expected files."""
     command_to_call, subscriber_config, publisher_config = read_config(config_file)
+    check_existing_files(publisher_config)
```
Comment (pnuu) [inline pytroll_runner/__init__.py:52]:
```suggestion
    old_files = check_existing_files(publisher_config)
```

File: pytroll_runner/__init__.py [hunk 3]
```diff
@@ -47,12 +49,22 @@ def parse_args(args=None):
...
+
     with closing(create_publisher_from_dict_config(publisher_config["publisher_settings"])) as pub:
         for _, mda in run_from_new_subscriber(command_to_call, subscriber_config):
             message = generate_message_from_expected_files(publisher_config, mda)
```
Comment (pnuu) [inline pytroll_runner/__init__.py:56]:
```suggestion
            message = generate_message_from_expected_files(publisher_config, old_files, extra_metadata=mda)
```

File: pytroll_runner/__init__.py [hunk 4]
```diff
@@ -91,16 +103,17 @@ def run_on_files(command, files):
 
 def generate_message_from_expected_files(pub_config, extra_metadata=None):
```
Comment (pnuu) [inline pytroll_runner/__init__.py:104]:
```suggestion
def generate_message_from_expected_files(pub_config, old_files, extra_metadata=None):
```

File: pytroll_runner/__init__.py [hunk 5]
```diff
@@ -91,16 +103,17 @@ def run_on_files(command, files):
...
     """Generate a message containing the expected files."""
-    filepattern = pub_config["expected_files"]
     metadata = populate_metadata(extra_metadata, pub_config.get("static_metadata", {}))
 
     dataset = []
-    files = glob(filepattern)
-    for filepath in sorted(files):
+
+    new_files = find_new_files(pub_config)
```
Comment (pnuu) [inline pytroll_runner/__init__.py:110]:
```suggestion
    new_files = find_new_files(pub_config, old_files)
```

File: pytroll_runner/__init__.py [hunk 6]
```diff
@@ -109,6 +122,14 @@ def generate_message_from_expected_files(pub_config, extra_metadata=None):
     return Message(pub_config["topic"], message_type, metadata)
 
 
+def find_new_files(pub_config):
+    """Find new files matching the file pattern."""
+    old_files = FILES_ALREADY_THERE
+    check_existing_files(pub_config)
+    new_files = FILES_ALREADY_THERE - old_files
+    return new_files
```
Comment (pnuu) [inline pytroll_runner/__init__.py:130]:
```suggestion
def find_new_files(pub_config, old_files):
    """Find new files matching the file pattern."""
    return check_existing_files(pub_config) - old_files
```

**Review by pnuu — CHANGES_REQUESTED** [review]
Some (untested) changes suggested to remove the global variable.

**Comment by mraspaud** [comment]
@pnuu I think I addressed your comments.

**Comment by mraspaud** [comment]
Ok, this support now regexes also.

File: pytroll_runner/__init__.py [hunk 7]
```diff
@@ -47,12 +50,31 @@ def parse_args(args=None):
...
+                message = generate_message_from_log_output(publisher_config, mda, log_output)
+            except KeyError:
+                message = generate_message_from_expected_files(publisher_config, mda, preexisting_files)
+                preexisting_files = check_existing_files(publisher_config)
             pub.send(message)
 
 
+def generate_message_from_log_output(publisher_config, mda, log_output):
+    """Generate message for the filenames present in the log output."""
+    new_files = re.findall(publisher_config["output_files_log_regex"], str(log_output))
```
Comment (pnuu) [inline pytroll_runner/__init__.py:67]: So if there's a regex pattern in the config, the output from the called script is parsed to catch the filenames? Handy.

Comment (mraspaud) [reply pytroll_runner/__init__.py:67]: yes exactly.

**Pull request merged**